iree/compiler/Conversion/LinalgToLLVM/Passes.cpp

```diff
@@ -117,6 +117,7 @@ void buildLLVMTransformPassPipeline(OpPassManager &passManager) {
         Shape::createMaterializeShapeCalculationsPass());
     nestedModulePM.addNestedPass<FuncOp>(
         Shape::createHoistShapeCalculationsPass());
+    nestedModulePM.addNestedPass<FuncOp>(createConvert1x1ConvToDotPass());
```

Comment:
Adding this here is problematic. We had the same issue with the DecomposeHLOClampPass. Long term HLO should not survive all the way into the executable translation. Could we move this pass before dispatch region creation so that the same behavior is preserved on the Linalg on tensors path as well.

Reply:
This cannot be moved to be before dispatch region creation because that will enable it for all backends. The purpose of this change is to give each backend more control and I'd like to not use it soon for GPU (at least for the convolution path I'm working on right now). Longer term in the Linalg-on-tensor-based solution, I'd actually expect this pass to go away or be replaced by `linalg.conv` to `linalg.matmul` equivalent so that each backend can then use that. So it's just for the old path I believe.

Reply:
Ok, just a note that we are building up quite a bit of such things, and we need to put them on either deprecation path, or adapt them to the linalg on tensors path. After the linalg on tensors stuff lands, I am going to do some major house-cleaning of all defunct backends passes.

Reply:
+100. I'd also like to start cleaning up things after landing the convolution work.